play-json/src/main/scala/play/api/libs/json/JsMacroImpl.scala

```diff
@@ -462,14 +462,17 @@ object JsMacroImpl {
 
     val syntaxImport = if (!multiParam && !writes) q"" else q"import $syntax._"
     @inline def buildCall = q"$canBuild.$applyOrMap(..${conditionalList(applyFunction, ApplyUnapply.unapplyFunction)})"
+    def readResult =
```

Comment:
If there is more than one parameter, then the Applicative instance for `Reads` is responsible of re-pathing, otherwise it needs to be done there